docs/concepts/registries.md

```diff
@@ -0,0 +1,82 @@
+
+There also exists the "registry registry", a registry that registers other registries. This registry registers registries with `ResourceLocation` names; additionally, it registers registries to `Class` keys. This allows one to look up the registry that registers a certain class (e.g. one can look up `Block.class` in the registry registry with `GameRegistry.findRegistry` to get the registry that registers blocks).
+
+Registering Things
+------------------
+
+The recommended way to register things is through the `RegistryEvent`s. In `RegistryEvent.NewRegistry`, registries should be created. Later, `RegistryEvent.Register` is fired once for each registered registry. Because `Register` is a generic event, the event handler should set the type parameter to the type of the object being registered. The event will contain the registry to register things to (`getRegistry`), and things may be registered with `register`(`All`) on the registry. Here's an example of an event handler that registers blocks:
+
+```java
+@SubscribeEvent
+public void registerBlocks(RegistryEvent.Register<Block> event) {
+    e.getRegistry().registerAll(block1, block2, ...);
+}
+```
+
+The order in which `RegistryEvent.Register` events fire is arbitrary, with the exception that `Block` will *always* fire first, and `Item` will *always* fire second, right after `Block`.
+
+!!! important
+    These events are fired *before* preinit. This means that `@Mod.EventBusSubscriber` (or `MinecraftForge.EVENT_BUS.register` in the `@Mod` class's constructor for e.g. Scala mods which do not support `static`) should be used to register the event handler before preinit.
+
+There is another, older way of registering objects into registries, using `GameRegistry.register`. Anytime something suggests using this method, it should be replaced with an event handler for the appropriate registry event. This method simply finds the registry corresponding to an `IForgeRegistryEntry` with `IFRE::getRegistryType`, and then registers the object to the registry. There is also a convenience overload that takes an IFRE and an RL, which is equivalent to `GameRegistry.register(ifre.setRegistryName(rl))`.
```

Comment:
It might not be clear that `IFRE::getRegistryType` is shorthand.

Also, I have no clue what you are referencing with "There is also a convenience overload that takes an IFRE and an RL, which is equivalent to `GameRegistry.register(ifre.setRegistryName(rl))`."